### PR TITLE
fix: add coordinator public key hash public input

### PR DESCRIPTION
**Pull request by 0xmad**
# Description

Add coordinator public key hash public input for MP circuits

## Additional Notes

Blocked by #1725 

## Related issue(s)

Related to #1614 

## Confirmation

- [x] I have read and understand MACI's [contributor guidelines](https://maci.pse.dev/docs/contributing) and [code of conduct](https://maci.pse.dev/docs/contributing/code-of-conduct).
- [x] I have read and understand MACI's [GitHub processes](https://github.com/privacy-scaling-explorations/maci/discussions/847).
- [x] I have read and understand MACI's [testing guide](https://maci.pse.dev/docs/testing).

### CODE REVIEW

File: circuits/circom/core/non-qv/processMessages.circom [hunk 1]
```diff
@@ -187,6 +189,9 @@ include "../../trees/incrementalQuinaryTree.circom";
     var derivedPubKey[2] = PrivToPubKey()(coordPrivKey);
     derivedPubKey === coordPubKey;
```
Comment (ctrlc03) [inline circuits/circom/core/non-qv/processMessages.circom:190]: you can get rid of this check and the `coordPubKey` signal - you are already checking that the the hash of the derived key is = to the hash that is on chain

File: core/ts/Poll.ts [hunk 2]
```diff
@@ -655,6 +655,11 @@ export class Poll implements IPoll {
 
     // ensure we pass the dynamic tree depth
     circuitInputs.actualStateTreeDepth = this.actualStateTreeDepth.toString();
+    const coordinatorPublicKey = this.coordinatorKeypair.pubKey;
```
Comment (ctrlc03) [inline core/ts/Poll.ts:658]: I think you can directly use the `PubKey` class method `hash` https://github.com/privacy-scaling-explorations/maci/blob/dev/domainobjs/ts/publicKey.ts#L86

File: cli/ts/commands/proveOnChain.ts [hunk 3]
```diff
@@ -214,11 +214,15 @@ export const proveOnChain = async ({
...
+    const [coordPubKeyHashOnChain, coordinatorPubKeyOnChain] = await Promise.all([
+      pollContract.coordinatorPubKeyHash(),
+      pollContract.coordinatorPubKey(),
+    ]);
+
     if (
       hashLeftRight(
-        BigInt((circuitInputs.coordPubKey as BigNumberish[])[0]),
-        BigInt((circuitInputs.coordPubKey as BigNumberish[])[1]),
+        BigInt((coordinatorPubKeyOnChain as BigNumberish[])[0]),
```
Comment (ctrlc03) [inline cli/ts/commands/proveOnChain.ts:224]: should it be instead `if (circuitInputs.coordintorPubKeyHash !== coordPubKeyHashOnChain.toString())`

File: contracts/tasks/helpers/Prover.ts [hunk 4]
```diff
@@ -156,13 +165,12 @@ export class Prover {
 
       this.validateCommitment(circuitInputs.currentSbCommitment as BigNumberish, currentSbCommitmentOnChain);
 
-      const coordPubKeyHashOnChain = BigInt(coordinatorPubKeyHash);
       const coordPubKeyHashOffChain = hashLeftRight(
-        BigInt((circuitInputs.coordPubKey as BigNumberish[])[0]),
-        BigInt((circuitInputs.coordPubKey as BigNumberish[])[1]),
+        BigInt((coordinatorPubKey as BigNumberish[])[0]),
```
Comment (ctrlc03) [inline contracts/tasks/helpers/Prover.ts:169]: this check is also redundant, will need to check circuitInputs.coordinatorPubKeyHash with the on chain one

File: contracts/tasks/helpers/Prover.ts [hunk 5]
```diff
@@ -80,16 +80,25 @@ export class Prover {
...
-      ]);
+    const [
+      treeDepths,
+      numSignUpsAndMessages,
+      numBatchesProcessed,
+      stateTreeDepth,
+      dd,
+      coordinatorPubKeyHash,
+      mode,
+      coordinatorPubKey,
```
Comment (kittybest) [inline contracts/tasks/helpers/Prover.ts:91]: Why we need both PubKeyHash and PubKey?

**Review by ctrlc03 — APPROVED** [review]
Thanks

**Pull request merged**